### PR TITLE
Added functionality for showing card with all conrgatulations

### PR DESCRIPTION
**Pull request by RomanKlishch**
- added servlet PreviewServlet;
- added method for return card from DB;
- added test;
- added templates

### CODE REVIEW

**Review by AlexKievDeveloper — APPROVED** [review]

**Comment by Rooman** [comment]
What does swowing mean?

File: src/test/java/com/greetingcard/web/servlet/card/EditCardServlet.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.greetingcard.web.servlet.card;
+
+import javax.servlet.http.HttpServlet;
+
+public class EditCardServlet extends HttpServlet {
```
Comment (Rooman) [inline src/test/java/com/greetingcard/web/servlet/card/EditCardServlet.java:5]: remove such code please

File: src/test/resources/testDB/migration/R__1setUpUsers.sql [hunk 2]
```diff
@@ -2,6 +2,8 @@ INSERT INTO users(firstName, lastName, login, email, password, salt, language_id
 VALUES ('admin','admin','admin', '@admin','2bb7998496899acdd8137fad3a44faf96a84a03d7f230ce42e97cd17c7ae429e','salt',1);
 INSERT INTO users(firstName, lastName, login, email, password, salt, language_id)
 VALUES ('user','user','user', '@user','8031377c4c15e1611986089444c8ff58c95358ffdc95d692a6d10c7b633e99df','salt',2);
+INSERT INTO users(firstName, lastName, login, email, password, salt, language_id)
+VALUES ('null','null','null', '@null','null','null',2);
```
Comment (Rooman) [inline src/test/resources/testDB/migration/R__1setUpUsers.sql:6]: null is such a bad choice for test data

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java [hunk 3]
```diff
@@ -106,6 +103,65 @@ public void createCard(Card card, User user) {
...
+                                .user(user)
+                                .card(card)
+                                .message(resultSet.getString("message"))
+                                .status(Status.getByNumber(resultSet.getInt("con_status")))
+                                .linkList(new LinkedList<>())
+                                .build();
+
+                        congratulationMap.put(congratulation_id, congratulation);
+                    }
+                    int link_id = resultSet.getInt("link_id");
```
Comment (Rooman) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java:146]: I'll cut your hands for such kind of mistakes. 
@AlexKievDeveloper You approved this.

**Review by Rooman — CHANGES_REQUESTED** [review]

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java [hunk 4]
```diff
@@ -106,6 +103,65 @@ public void createCard(Card card, User user) {
...
+                    if (link_id != 0) {
+                        Link link = Link.builder()
+                                .id(link_id)
+                                .link(resultSet.getString("message"))
+                                .type(LinkType.getByNumber(resultSet.getInt("type_id")))
+                                .build();
+                        congratulationMap.get(congratulation_id).getLinkList().add(link);
+                    }
+                }
+                card.setCongratulationList(new LinkedList<>(congratulationMap.values()));
```
[the remark this answers is not in the document]
Comment (RomanKlishch) [reply src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java:156]: Я использую эту колекцию только для добавления новой линки в конец списка

**Review by AlexKievDeveloper — APPROVED** [review]

**Review by OleksandrGubariev — APPROVED** [review]

**Pull request merged**